### ISSUE TITLE
Building fails on Arch Linux if ROOT is installed from Arch User Repository

### ISSUE THREAD

**Issue by pdigiglio** [bat/bat#155]
I am trying to build BAT on Arch Linux but the compilation fails due to some undefined references. Some relevant information may be:

### Building process

* ROOT (v. 6.06/02) has been installed from the [Arch User Repository](https://aur.archlinux.org/packages/root/);

* The _Makefile_ is generated with the command
```
% CFLAGS="-march=native -O2 -pipe -fstack-protector-strong" CXXFLAGS="${CFLAGS}" CPPFLAGS="${CFLAGS}" ./configure --enable-parallel --with-gnu-ld --with-rootsys=/usr 
```
Everything seems to work fine so far and, among other output lines, I get the following:
```
checking for root-config... /usr/bin/root-config
checking for root... /usr/bin/root
checking for rootcint... /usr/bin/rootcint
checking for rootcling... /usr/bin/rootcling
checking for rlibmap... /usr/bin/rlibmap
checking whether ROOT version >= 5.34.14... yes
checking whether ROOT is compiled with MathMore support... yes
```
Also, the include and library paths in the _Makefile_ look correct.

* When I `make` the project, the compilation fails at this point:

```
../../../../src/.libs/libBAT.so: undefined reference to `TROOT::RegisterModule(char const*, char const**, char const**, char const*, char const*, void (*)(), std::vector<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int>, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> > > const&, char const**)'
../../../../models/base/.libs/libBATmodels.so: undefined reference to `TString::TString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
Makefile:601: recipe for target 'runSystematics' failed
```

### Additional info
 * The error occurs both if I set the enviroment variable `ROOTSYS=/usr` (with `. bin/thisroot.sh`) and if I do not.
 * I was able to compile BAT by downloading and building the ROOT source from the [CERN website](https://root.cern.ch/content/release-60602).

**Comment by fredRos** [comment]
A linker error is always nasty. I have a few ideas:


> ./configure --enable-parallel --with-gnu-ld

what if you try without `--with-gnu-ld`?

It sounds like you played around a lot with root, are you sure you have only one version of root available in your system? i suspect you specify one version of root during `configure` but another version is linked that misses the mentioned method. 

if you retry do build bat in a clean environment (i.e., only one version of  `root`), do you get the same error?

**Comment by pdigiglio** [comment]
> if you retry do build bat in a clean environment (i.e., only one version of `root`), do you get the same error?

Yes, I got this error on the very first try when the only ROOT version installed was the one provided by the Arch User Repository.

I'm still trying to inspect what's exactly going on but I think the problem is related to the compiler. Arch Linux provides GCC v5.3.0 and [this](https://root.cern.ch/phpBB3/viewtopic.php?t=20587#p88838) post on the ROOT forum (Nov. 2015) says
> Unfortunately we do not support yet GCC version >= 5.1. The reason is that the LLVM/clang we use for the interpreter does not supper the new GCC ABI. Thus the missing symbol you encountered. We will integrate a new version of LLVM/clang as soon as possible.

As a matter of fact:
 1. I just realized that the ROOT I compiled myself from the sources provided by the CERN website has linking problems too (it was a little tricky as the errors don't always show up). 
 2. The Arch User Repository script to build ROOT uses the compilation flag `-D_GLIBCXX_USE_CXX11_ABI=0`. I was able to compile BAT adding this flag as well in the `CFLAGS` and `CXXFLAGS` variables.

Thanks for your help! I hope someone else can benefit from this thread.
